### ISSUE TITLE
requests - karma coverage, jshint, bundle multiple javascripts into one js file

### ISSUE THREAD

**Issue by golgistudio** [vigetlabs/gulp-starter#218]
Hello,

Thank you for making the gulp-starter available!   I am using it on the following front end demo project:
[github:](https://github.com/golgistudio/d3-scatter-plot)
[running version is here: ](http://www.golgistudio.com/d3-scatter-plot)

I am still trying to figure out how to get the development and production builds working.  It executes and substitutes the files fine, but I get a number of errors when it runs, but I'll figure it out - that's not my question/request.

 I wasn't sure if you were accepting external requests, so am bundling it all into one - let me know if you'd like me to enter them individually.

Here are my requests....

1) JShint - I've added jshint gulp task and have it running on travis, but would like travis to fail if there are any warnings or error.  

2) Karma coverage - I added karma coverage, and a report is output, but no coverage results are included and I don't see it being kicked off during the build process.

3) Bundle into one javascript.   It is great that you support including different javascript files on different html pages, but I would like to be able to bundle a set of javascripts into one javascript file to be included in addition to support for different javascript files in different html files.

4) I include a reference to an image file in my javascript.  I would like a clean way to substitute it with the hashed name.

Thank you!

**Comment by greypants** [comment]
> would like travis to fail if there are any warnings or error.
you'll have to figure out a way to make gulp exit with a status of 1. This thread migiht help https://github.com/sindresorhus/jshint-stylish/issues/6

>  Karma coverage 
Can't really help you there as I haven't worked with that before. Note that I'm not running Karma through gulp. It's run separately from the command line through `npm run test`, which runs the `gulp production task` (as a dumb test to see if everything compiles without erroring and then runs `karma start --single-run`

>  I would like to be able to bundle a set of javascripts into one javascript file
This bit of the config is where you specify what files you want to output: https://github.com/vigetlabs/gulp-starter/blob/master/gulpfile.js/config.json#L18. A file is output for each key/value pair. https://github.com/vigetlabs/gulp-starter/blob/master/gulpfile.js/config.json#L17, which creates an additional file with shared dependencies. The key is used to name the file that gets output, and the value is an array of entry points. This can be more than one file. Just add any additional files to the array.

> I include a reference to an image file in my javascript. I would like a clean way to substitute it with the hashed name.

This should already be happening. It happens in the rev tasks that get run witih `npm run production`: https://github.com/vigetlabs/gulp-starter/blob/master/gulpfile.js/tasks/rev/rev-update-references.js There is an example of it working in the [demo](http://code.viget.com/gulp-starter/) if you crack open the console: 
![image](https://cloud.githubusercontent.com/assets/647549/11051611/91bf2e6c-871e-11e5-9182-6c2ca5fa4272.png)

Hope some of that was helpful. Good luck!

**Comment by golgistudio** [comment]
Thank you!  

~ JSHint - I'll look into this.  Thank you for the link.
~ Karma coverage - thank you.  I'll look into it more myself.  I tried replacing the karma with gulp karma to try to get the coverage, but backed that out.
~ Bundling javascripts - I fixed this on my end.  I wasn't using import/export for the dependencies, but I am now.  
~ Updating references in javascript - yes - I realize now it is already working.

Thanks again - really terrific set of tools.  Works beautifully!
